presto/eval.py

```diff
@@ -426,21 +428,21 @@ def finetune(self, pretrained_model) -> PrestoFineTuningModel:
     def finetuning_results(
         self,
         pretrained_model,
-        model_modes: List[str],
+        sklearn_model_modes: List[str],
     ) -> Tuple[Dict, Optional[PrestoFineTuningModel]]:
-        for model_mode in model_modes:
-            assert model_mode in ["Regression", "Random Forest", "finetune"]
+        for model_mode in sklearn_model_modes:
+            assert model_mode in ["Regression", "Random Forest", "CatBoostClassifier"]
 
         results_dict = {}
-        finetuned_model: Optional[PrestoFineTuningModel] = None
-        if "finetune" in model_modes:
-            finetuned_model = self.finetune(pretrained_model)
-            results_dict.update(self.evaluate(finetuned_model, None))
-            if self.spatial_inference_savedir is not None:
-                self.spatial_inference(finetuned_model, None)
-
-        sklearn_modes = [x for x in model_modes if x != "finetune"]
-        if len(sklearn_modes) > 0:
+        # we want to always finetune the model, since the sklearn models
```

Comment:
So for my understanding, the biggest change here is that previously you'd have the `finetune` mode in which presto was finetuned on as supervised task and next evaluated using a `sklearn` model, while otherwise the `sklearn` model was directly fitted on the pretrained presto. And now for WorldCereal eval task we always finetune presto first before evaluating the resulting embeddings for a sklearn or catboost downstream classification task?

Reply:
yes, that's right. This is to better reflect the approach we will ultimately adopt.